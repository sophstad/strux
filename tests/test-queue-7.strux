int main() {
      Queue::num qn = new Queue::num();
      qn.add(3.6);
      qn.add(4.216);
      qn.add(-5.25);
      qn.add(6.4);
      qn.show();

      Queue::int qi = new Queue::int();
      qi.add(23623);
      qi.add(-39632);
      qi.add(4 * 1);
      qi.add(356);
      qi.remove();
      qi.add(89494);
      qi.add(22);
      qi.show();

      Queue::string qs = new Queue::string();
      qs.add("Josh");
      qs.add("Kofi");
      qs.add("Millie");
      qs.add("Sophie");
      int qsSize = qs.size();
      print(qsSize);
      qs.show();

      Queue::bool qb = new Queue::bool();
      qb.add(true);
      qb.add(false);
      qb.show();

      return 0;
}
