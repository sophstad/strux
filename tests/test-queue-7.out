+----------+----------+-----------+----------+
| 3.600000 | 4.216000 | -5.250000 | 6.400000 |
+----------+----------+-----------+----------+
Head                                      Tail
+--------+---+-----+-------+----+
| -39632 | 4 | 356 | 89494 | 22 |
+--------+---+-----+-------+----+
Head                         Tail
4
+------+------+--------+--------+
| Josh | Kofi | Millie | Sophie |
+------+------+--------+--------+
Head                         Tail
+---+---+
| 1 | 0 |
+---+---+
Head Tail
